service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -231,14 +232,47 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
   @Override
   public ResponseEntity<AbortRunSetResponse> abortRunSet(UUID runSetId) {
     AbortRunSetResponse aborted = new AbortRunSetResponse();
-    List<UUID> dummyRunsList = new ArrayList<>();
+    List<String> errors = new ArrayList<>();
 
-    dummyRunsList.add(UUID.randomUUID());
-    dummyRunsList.add(UUID.randomUUID());
+    // Get the run set associated with runSetId
+    RunSet runSet = runSetDao.getRunSet(runSetId);
+    // Update the run set to have a canceling state
+    runSetDao.updateStateAndRunDetails(
+        runSetId,
+        CbasRunSetStatus.CANCELING,
+        runSet.runCount(),
+        runSet.errorCount(),
+        OffsetDateTime.now());
 
     aborted.runSetId(runSetId);
+
+    // Get a list of workflows able to be canceled
+    List<Run> runningWorkflows =
+        runDao.getRuns(new RunDao.RunsFilters(runSetId, NON_TERMINAL_STATES));
+    List<UUID> submittedAbortWorkflows = new ArrayList<>();
+
+    for (Run run : runningWorkflows) {
+      // Trying inside the for-loop in case a single run fails to be updated
+      try {
+        cromwellService.cancelRun(run);
+        submittedAbortWorkflows.add(run.runId());
+        // Update each run to have a 'canceling' run state
+        runDao.updateRunStatus(run.runId(), CbasRunStatus.CANCELING, OffsetDateTime.now());
+      } catch (cromwell.client.ApiException e) {
+        String msg = "Unable to abort workflow %s.".formatted(run.runId());
```

Comment:
nit: Adding this message prefix `Unable to abort workflow` here will make it repetitive for each run. So if 3 runs had exceptions the final error message that user will see would be
``` 
Run set canceled with errors: Unable to abort workflow X. Unable to abort workflow Y. Unable to abort workflow Z
```
To make it DRY this prefix message can be appended on L271. Something like:
```
Run set canceled with errors. Unable to abort workflow(s): X, Y, Z
```